Removido funções de cadastro e atualização de usuário do domínio principal

GCastilho opened the pull request:
O sistema de cadastro e atualização de senha dos usuários ainda é feita no domínio principal do site, isso deve ser trocado para funcionar pela API (e os handlers antigos removidos)

- [x] Implementar a função de cadastro de usuários na API
- [x] Implementar a função de alteração de senha na API
- [x] Migrar os testes que ainda estão no domínio principal para a API
- [x] Migrar o registro de usuários do cliente para utilizar os métodos da API
- [x] Migrar o sistema de atualização de senha do cliente para utilizar o método da API
- [x] Refatorar o módulo axios do cliente e utilizar axios.create
- [x] Remover TODOS os handlers dos domínio principal

GCastilho committed:
Implementado testes na autenticação da API

-- Implemenado testes em /v1/user/authenticate e /v1/user/deauthenticate
-- Corrigido um bug no handler de desautenticação
-- Removido testes de login do domínio principal

GCastilho committed:
Adicionado registro de usuários na HttpApi

-- Adicionado um handler para registro de usuários na httpApi em POST /v1/user
-- Movido testes do domínio principal para o api

GCastilho committed:
Adicionado handler de atualizar senha na API

-- Adicionado um handler PATCH /v1/user/password que atualiza a senha do usuário
-- Movido os testes da /user/changepass para a httpApi
-- Removido o arquivo de testes do domínio principal, pois ele não tem mais testes

GCastilho committed:
Bugfix do master

-- Corrigido um bug onde os módulos do database e da currencyApi não estavam sendo carregados na ordem correta que resultava na referência da CurrencyApi do model da person ser um objeto vazio

GCastilho commented:
@joaojvds Vc poderia fazer a parte do frontend?

GCastilho commented on src/server/api/v1/user.ts:
```diff
@@ -166,6 +191,32 @@ router.get('/', (_req, res) => {
 	})
 })
 
+router.patch('/:password', async (req, res): Promise<any> => {
```
O parâmetro password não est sendo usado, dá pra tirar o :

joaojvds committed:
Registros de usuarios usam os metodos da API

-- Migrado o registro de usuários do cliente para utilizar os métodos da API

joaojvds committed:
Mudança de senha é feita pelos métodos da API

-- Migrado o sistema de atualização de senha do cliente para utilizar o método da API
-- alterado header para permitir requests por PATCH

joaojvds commented:
Porque senão ele vai fazer post na porta 3000, não dando para testar usando o `npm run dev`, quando a gente tiver que compilar ai gente muda para host

GCastilho commented:
Tem um roteador na porta 3000 que redireciona para a 3001. Se isso não tá acontecendo é o roteador que não está funcionando

GCastilho commented:
Mas eu tive uma ideia melhor então pode deixar assim por enquanto

joaojvds commented:
Então tem que da uma olhada nisso porque se você deixar só como host ele vai da 404

GCastilho commented:
Darei. Deve ser um erro com o roteador

GCastilho commented on src/server/api/v1/user.ts:
```diff
@@ -191,17 +191,17 @@ router.get('/', (_req, res) => {
-		await user.checkPassword(req.body.old)
-		await user.changePassword(req.body.new)
+		await user.checkPassword(req.body.oldPassword)
+		await user.changePassword(req.body.newPassword)
```
Desfaça tudo isso, não era pra ter mexido no servidor, @joaojvds

GCastilho commented on client/src/routes/user/changepass.svelte:
```diff
@@ -82,9 +83,9 @@
 	{/if}
-	<FancyInput id="password_old" type="password" autocomplete="current-password">Old Password</FancyInput>
-	<FancyInput id="password_new" type="password" autocomplete="new-password">New Password</FancyInput>
-	<FancyInput id="password_confirm" type="password" autocomplete="new-password">Confirm Password</FancyInput>
+	<FancyInput id="password_old" type="password">Old Password</FancyInput>
+	<FancyInput id="password_new" type="password">New Password</FancyInput>
+	<FancyInput id="password_confirm" type="password">Confirm Password</FancyInput>
```
Desfaça isso tbm

joaojvds committed:
fix

GCastilho committed:
Removido requests não utilizados do domínio principal

-- Removido os requests de cadastro, login e alteração de senha do domínio principal
-- Corrigido um teste que ainda usava o /login para fazer login
-- Corrigido um bug leve em um componente do svelte

GCastilho committed:
Merge branch 'master' into api-register

joaojvds committed:
Adicionado custom config no axios

joaojvds committed:
removidos comentários em relação as portas

GCastilho commented on client/src/routes/login.svelte:
```diff
@@ -1,5 +1,5 @@
 <script>
-	import axios from 'axios'
+	import axios from "../utils/axios"
```
Troca pra aspas simples

GCastilho commented on client/src/routes/login.svelte:
```diff
@@ -21,9 +21,8 @@
-				`${location.protocol}//api.${location.hostname}:3001/v1/user/authentication`,
-				{ email, password },
-				{ withCredentials: true }
+				'/v1/user/authentication',
+				{ email, password }
```
Em uma linha só isso n cabe? (passa dos 80 caracteres?)

joaojvds replied:
os que deu para colocar em uma linha eu coloquei, os que estão assim é pq ultrapassou o limite

GCastilho reviewed (commented):
detalhes

joaojvds committed:
detalhes

GCastilho committed:
Consertado porta do axios

-- O módulo do axios agora manda para a porta 3001 se em dev e para a 3000 se não

joaojvds reviewed (approved):
ta tudo certo
